### PR TITLE
Add unit test for filter command

### PR DESCRIPTION
**Pull request by soakedumplings**
### Add comprehensive unit test for filter command:

- **Functionality:** Accurately filters (0, 1 or multiple companies)
- **Edge Case:** empty address book, case-insensitive statuses, does not modify original address book, null model, null status 
- **Others**: equa()l method, toString() method

Note:
- Previous version allowed null status and throw error only when execute() is called.
- Current implementation catches null status input early and throws exception before execute().

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/FilterCommandTest.java [hunk 1]
```diff
@@ -56,6 +67,17 @@ public void execute_matchingStatus_singleCompanyFound() {
         assertEquals(Arrays.asList(CONS), model.getFilteredCompanyList());
     }
 
+    // Ensures filtering finds the correct company with given status (multiple companies found)
```
Comment (kzfsg) [inline src/test/java/seedu/address/logic/commands/FilterCommandTest.java]: Javadoc comment format for tests!

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 2]
```diff
@@ -22,7 +22,13 @@ public class FilterCommand extends Command {
 
     private final Status status;
 
+    /**
```
Comment (kzfsg) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java]: good

File: src/test/java/seedu/address/logic/commands/FilterCommandTest.java [hunk 3]
```diff
@@ -65,12 +87,56 @@ public void execute_noCompanyWithStatus_zeroFound() {
         assertEquals(Collections.emptyList(), model.getFilteredCompanyList());
     }
 
+    // Verifies filtering doesn't mutate the model's address book contents.
```
Comment (kzfsg) [inline src/test/java/seedu/address/logic/commands/FilterCommandTest.java]: javadoc comments man
